packages/dev/core/src/Meshes/Compression/dracoCompressionWorker.ts

```diff
@@ -1,48 +1,154 @@
+        const availableBytes = destination ? destination.byteLength : 0;
+        const buffer = availableBytes < encodedLength ? new ArrayBuffer(encodedLength) : destination!;
+        const encodedData = new Int8Array(buffer, 0, encodedLength);
+        for (let i = 0; i < encodedLength; i++) {
+            encodedData[i] = encodedNativeBuffer.GetValue(i);
+        }
+
+        return { data: encodedData, attributeIDs: attributeIDs };
+    } finally {
+        if (mesh) {
+            encoderModule.destroy(mesh);
+        }
+        if (meshBuilder) {
+            encoderModule.destroy(meshBuilder);
+        }
+        if (encoder) {
+            encoderModule.destroy(encoder);
+        }
+        if (encodedNativeBuffer) {
+            encoderModule.destroy(encodedNativeBuffer);
+        }
+    }
 }
 
-interface AttributeMessage extends AttributeData {
-    id: "attribute";
+/**
+ * The worker function that gets converted to a blob url to pass into a worker.
+ * To be used if a developer wants to create their own worker instance and inject it instead of using the default worker.
+ */
+export function EncoderWorkerFunction(): void {
+    let encoderPromise: PromiseLike<EncoderModule> | undefined;
+
+    onmessage = (event) => {
+        const message = event.data;
+        switch (message.id) {
+            case "init": {
+                // if URL is provided then load the script. Otherwise expect the script to be loaded already
+                if (message.url) {
+                    importScripts(message.url);
+                }
+                const initDecoderObject = message.wasmBinary ? { wasmBinary: message.wasmBinary } : {};
+                encoderPromise = DracoEncoderModule(initDecoderObject);
+                postMessage({ id: "initDone" });
+                break;
+            }
+            case "encodeMesh": {
+                if (!encoderPromise) {
+                    throw new Error("Draco encoder module is not available");
+                }
+                encoderPromise.then((encoder) => {
+                    const result = EncodeMesh(encoder, message.attributes, message.indices, message.options, message.buffer);
+                    postMessage({ id: "encodeMeshDone", encodedMeshData: result }, result ? [result.data.buffer] : undefined);
+                });
+                break;
+            }
+        }
+    };
 }
-export type Message = InitDoneMessage | DecodeMeshDoneMessage | IndicesMessage | AttributeMessage;
-// WorkerGlobalScope
-declare function importScripts(...urls: string[]): void;
-declare function postMessage(message: Message, transfer?: any[]): void;
 
 /**
  * @internal
  */
-export function decodeMesh(
-    decoderModule: any /*DecoderModule*/,
+export function DecodeMesh(
+    decoderModule: any,
     data: Int8Array,
-    attributes: { [kind: string]: number } | undefined,
+    attributeIDs: { [kind: string]: number } | undefined,

```

Comment:
This can also be:
```suggestion
    attributeIDs: Record<string, number> | undefined,
```
`Record` is a TS utility type for this scenario.

Reply:
Oooh, this'll also be useful for my `quantizationBits` option. Thanks.